core/sql/generator/Generator.cpp

```diff
@@ -1737,6 +1737,82 @@ TrafDesc * Generator::createConstrKeyColsDescs(Int32 numKeys,
   return first_key_desc;
 }
 
+// ****************************************************************************
+// This method creates a set of trafodion descriptors (TrafDesc) based on 
+// ComTdbVirtTablePrivInfo 
+//
+// see ComTdb.h for a description of the ComTdbVirtTablePrivInfo
+// see TrafDDLdesc.h for a description of TrafDesc for the priv_desc
+// ****************************************************************************
+TrafDesc * Generator::createPrivDescs( const ComTdbVirtTablePrivInfo * privInfo,
+                                       Space * space)
+{
+  // When authorization is enabled, each object must have at least one grantee
+  // - the system grant to the object owner
+  std::vector<PrivMgrDesc> *privGrantees = privInfo[0].privmgr_desc_list;
+  DCMPASSERT (privGrantees.size() > 0);
+ 
+  TrafDesc * priv_desc = TrafAllocateDDLdesc(DESC_PRIV_TYPE, space);
+  TrafDesc * first_grantee_desc = NULL;
+  TrafDesc * prev_grantee_desc = NULL;
+
+  // generate a TrafPrivGranteeDesc for each grantee and
+  // attach to the privileges descriptor (priv_desc)
+  for (int i = 0; i < privGrantees->size(); i++)
+    {
+      PrivMgrDesc granteeDesc = (*privGrantees)[i];
```

Comment:
Is there a need to make a copy of the object here? Or would it be enough to make this a reference?